dts/xtensa/intel/intel_adsp_ace15_mtpm.dtsi

```diff
@@ -25,17 +25,23 @@
 			device_type = "cpu";
 			compatible = "cdns,tensilica-xtensa-lx7";
 			reg = <1>;
-			cpu-power-states = <&d0i3 &d3>;
+			cpu-power-states = <&active &d0i3 &d3>;
 		};
 
 		cpu2: cpu@2 {
 			device_type = "cpu";
 			compatible = "cdns,tensilica-xtensa-lx7";
 			reg = <2>;
-			cpu-power-states = <&d0i3 &d3>;
+			cpu-power-states = <&active &d0i3 &d3>;
 		};
 
 		power-states {
+			active: active {
+				compatible = "zephyr,power-state";
+				power-state-name = "active";
+				min-residency-us = <0>;
+				exit-latency-us = <0>;
+			};
```

Comment:
I particularly don't think that is a good thing. Declare this state here is to workaround the problem.

I think one way to approach the problem without need the custom data is creating a proper state (runtime-idle with some substate, or any other state ...) with minimal residency. Then the application can put/release locks in this state if it wants to switch clock on idle.

Another possibility and clear (?) is adding a proper API in the adsp clock control (look other comments)